### PR TITLE
Lazy-load the JupyterLite embedded page in documentation

### PR DESCRIPTION
**Pull request by HaudinFlorence**
Remove the retrolite directive in the documentation, since there are already the links and the buttons to click and make an online try.

### CODE REVIEW

**Comment by jasongrout** [comment]
Why would you like to remove the embedded example? For example, is it to simplify the page visually? To make it easier to download?

I think there is a far higher engagement if we embed an example directly in the page, rather than making people click a link. I'm not sure how we'd be able to track that, but I think it's worth embedding an example *and* providing a link to many more examples. Thoughts?

**Comment by HaudinFlorence** [comment]
Thanks for your reply. The main purpose of proposing this change was to make the download easier when a user only wants to have a look at the doc. Also to simplify the page visually (here the index is pushed at the bottom of the page and not really visible anymore). Perhaps we can think about embedding examples another place and redistribute things differently ?

File: docs/source/index.rst [hunk 1]
```diff
@@ -15,10 +16,13 @@ Try it online
...
    :target: ./lite/lab?path=ipyleaflet.ipynb
 
-You can try ipyleaflet below, or open many other live examples in a new browser tab with
-`JupyterLite <./lite/lab?path=ipyleaflet.ipynb>`_ or `RetroLite <./lite/retro/tree>`_.
+You can try ipyleaflet below, or open many other live examples in a new browser tab with : `JupyterLite <./lite/lab?path=ipyleaflet.ipynb>`_ or `RetroLite <./lite/retro/tree>`_.
 
 .. retrolite:: ipyleaflet.ipynb
+   :width: 100%
+   :height: 810px
+   :prompt: Try Retrolite!
```
Comment (martinRenou) [inline docs/source/index.rst:24]:
```suggestion
   :prompt: Try ipyleaflet!
```

**Review by martinRenou — COMMENTED** [review]
Thanks! I just have a minor suggestion :)

File: docs/source/index.rst (hunk 1 above)
Comment (HaudinFlorence) [reply docs/source/index.rst]: Thanks for the suggestion. This has been changed.

**Comment by martinRenou** [comment]
Thank you!

**Comment by jasongrout** [comment]
> The main purpose of proposing this change was to make the download easier when a user only wants to have a look at the doc. 

My main purpose in reintroducing the example on the front page is to make it much more likely that a user will interact with it. The thought is that every click you require a user to take really cuts down on engagement, and if you make it *so* easy to try things, the user actually will.

I would be very interested in data that could validate that assumption, i.e., how many users will run code in an embedded retrolite, vs how many users will click through to a separate page. My guess is that far more users will run code in an embedded retrolite than will click through to a separate page to run code. However, I think other reasons prevent us from collecting such data.

> Also to simplify the page visually (here the index is pushed at the bottom of the page and not really visible anymore).

What do you think of putting the embedded retrolite behind an HTML `details` element, so that it doesn't take up the vertical space, but still is immediately available to users? I think that takes care of the visual point you mention. Something like:

<details><summary>Try ipyleaflet now</summary>
[The embedded retrolite iframe]
</details>

**Comment by jasongrout** [comment]
It looks like the HTML `details` element has an [event](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#events) for when it is opened, so we could just have a details element that, the first time it is opened, creates the necessary iframe.

**Comment by jasongrout** [comment]
Thanks for introducing the prompt option in this PR. I didn't know about that option, which does seem to solve at least one of the concerns about a heavy page. Awesome!

**Comment by HaudinFlorence** [comment]
> Thanks for introducing the prompt option in this PR. I didn't know about that option, which does seem to solve at least one of the concerns about a heavy page. Awesome!

This was a suggestion of @martinRenou . I did not know about it neither

**Pull request merged**